core/validation/validate.py

```diff
@@ -131,20 +132,21 @@ def validate_types(
                            expected data types.
     :return: A list of wrong type failures, if any.
     """
-    sheet_types = workbook[sheet_name].dtypes
-
     wrong_type_failures = []
-    for column, exp_type in column_to_type.items():
-        if column in sheet_types.keys():
-            got_type = sheet_types[column]
+    for index, row in workbook[sheet_name].iterrows():
+        for column, exp_type in column_to_type.items():
+            got_value = row.get(column)
+            if got_value is None or pd.isna(got_value):
+                continue
+            got_type = _PY_TO_NUMPY_TYPES.get(type(got_value).__name__, type(got_value).__name__)
```

Comment:
This is what I'll do as I suppose the np datatypes are now redundant

Reply:
After a bit of investigation, it looks like this change prompts other changes in the codebase in the schema and how it is parsed, and would require changes to all of the testing relating to the schema. Simply removing this dictionary look-up breaks the code, and as it would require a medium-sized ticket in itself (if not a larger one), we're going to put this in the backlog as tech debt and this draft PR is now the final PR.